src/stan/mcmc/hmc/nuts/adapt_auto_e_nuts.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN_MCMC_HMC_NUTS_ADAPT_AUTO_E_NUTS_HPP
+#define STAN_MCMC_HMC_NUTS_ADAPT_AUTO_E_NUTS_HPP
+
+#include <stan/callbacks/logger.hpp>
+#include <stan/mcmc/hmc/nuts/auto_e_nuts.hpp>
+#include <stan/mcmc/hmc/mpi_cross_chain_adapter.hpp>
+#include <stan/mcmc/mpi_auto_adaptation.hpp>
+
+namespace stan {
+namespace mcmc {
+/**
+ * The No-U-Turn sampler (NUTS) with multinomial sampling
+ * with a Gaussian-Euclidean disintegration and adaptive
+ * dense or diagonal metric and adaptive step size
+ */
+template <class Model, class BaseRNG>
+class adapt_auto_e_nuts : public auto_e_nuts<Model, BaseRNG>,
+			  public mpi_cross_chain_adapter<adapt_auto_e_nuts<Model, BaseRNG>>,
+			  public stepsize_covar_adapter {
+protected:
+  const Model& model_;
+public:
+  adapt_auto_e_nuts(const Model& model, BaseRNG& rng)
+    : model_(model), auto_e_nuts<Model, BaseRNG>(model, rng),
+    stepsize_covar_adapter(model.num_params_r()) {}
+
+  ~adapt_auto_e_nuts() {}
+
+  sample
+  transition(sample& init_sample, callbacks::logger& logger) {
```

Comment:
I've updated the call to mpi adapter, see https://github.com/stan-dev/stan/blob/117664b0875c83f2d2c66eb3a2ca028d84928991/src/stan/mcmc/hmc/nuts/adapt_dense_e_nuts.hpp#L34. Its logic is easier to understand. It also makes single chain default back to regular warmup.